### PR TITLE
increase dependencies

### LINKED ISSUE #98: Increase dependencies
Symfony 7 is about to be released (in one month, more or less), so it's time to consider upgrading minimum requirements.

Here's my proposal:
* drop Symfony 4 (it's EOL)
* drop PHP 7.4 and 8.1 (the former is EOL, the latter will be EOL in a few weeks)
* allow Symfony 7 (it will require to allow dev deps in CI, at least until the release)
* increase minor version for other deps (for vimeo, we could also upgrade to major)

The new PHP requirement will involve also upgrading our CS rules.

### PR DESCRIPTION
**Pull request by garak**
Fix #98

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
 
     strategy:
       matrix:
-        container: [ "php74", "php80", "php81", "php82" ]
+        container: [ "php81", "php82" ]
```
Comment (antonkomarev) [inline .github/workflows/php.yml:23]: Do we need to add 8.3 here?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -36,10 +36,10 @@ composer behat
 
 ```bash
 docker-compose up --build -d
-docker-compose exec php74 composer install
-docker-compose exec php74 composer phpspec
-docker-compose exec php74 composer behat
-docker-compose exec php74 composer php-cs-fixer
+docker-compose exec php82 composer install
```
Comment (antonkomarev) [inline CONTRIBUTING.md:39]: Why not 8.3 (highest) or 8.1 (lowest)

File: docker-compose.yml [hunk 3]
```diff
@@ -1,22 +1,6 @@
...
 
-  php74:
-    build:
-      context: .docker/development/php74
-    working_dir: /application
-    stdin_open: true
-    volumes:
-      - .:/application:cached
-
-  php80:
```
Comment (antonkomarev) [inline docker-compose.yml:12]: We removed 8.0 and 7.4, but didn't added 8.3

File: .github/workflows/php.yml (hunk 1 above)
Comment (garak) [reply .github/workflows/php.yml:23]: The answer to this question (and to the following ones) is in my comment above: we still don't have a docker image for PHP 8.3

File: docker-compose.yml (hunk 3 above)
Comment (garak) [reply docker-compose.yml:12]: I managed to create the new image, so now everything includes PHP 8.3 as well

**Review by antonkomarev — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+* add compatibility to Symfony v7 and PHP 8.3
+* drop compatibility to Symfony v4 and PHP 7.4
```
Comment (JellyBellyDev) [inline CHANGELOG.md:12]: we shouldn't label it as braking?

File: composer.json [hunk 5]
```diff
@@ -60,10 +60,12 @@
         "phpspec": "bin/phpspec run --format=pretty --ansi",
         "phpspec-coverage": "bin/phpspec run -f progress -c phpspec-coverage.yml --ansi",
         "behat": "bin/behat --snippets-for",
-        "docker:build:php74": "docker build -t pugx/poser:php74 -f .docker/base/php74/Dockerfile .",
-        "docker:push:php74": "docker push pugx/poser:php74",
-        "docker:build:php80": "docker build -t pugx/poser:php80 -f .docker/base/php80/Dockerfile .",
-        "docker:push:php80": "docker push pugx/poser:php80"
+        "docker:build:php81": "docker build -t pugx/poser:php81 -f .docker/base/php81/Dockerfile .",
+        "docker:push:php81": "docker push pugx/poser:php81",
+        "docker:build:php82": "docker build -t pugx/poser:php82 -f .docker/base/php80/Dockerfile .",
```
Comment (JellyBellyDev) [inline composer.json:65]: there is a typo on `base/php80` => `base/php82`

File: composer.json [hunk 6]
```diff
@@ -60,10 +60,12 @@
...
+        "docker:push:php82": "docker push pugx/poser:php82",
+        "docker:build:php83": "docker build -t pugx/poser:php83 -f .docker/base/php80/Dockerfile .",
```
Comment (JellyBellyDev) [inline composer.json:67]: there is a typo on `base/php80` => `base/php83`

**Review by JellyBellyDev — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (garak) [reply CHANGELOG.md:12]: I don't see mentions of BC breaks in our changelog.
I guess we can simply publish the new release as a major version (v3.0.0), the breaking is implicit

Comment (JellyBellyDev) [reply CHANGELOG.md:12]: 👍

**Review by JellyBellyDev — APPROVED** [review]
LGTM

**Pull request merged**